mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -109,6 +137,9 @@ impl<E: std::error::Error + Send + Sync> PartQueueProducer<E> {
             trace!("closed channel");
         } else {
             self.bytes_sent.fetch_add(part_len, Ordering::SeqCst);
+            let _ = self
+                .next_read_window_offset
```

Comment:
do we actually need to receive `next_read_window_offset` from the client? don't we have enough information in `Prefetcher` / `RequestTask` to compute that value? 

having to pass `next_read_window_offset` from `ObjectPartStream` to `Prefetcher` through multiple layers so it can just do:
```rust
let next_read_window_offset = current_task.next_read_window_offset();
self.prepare_next_read_window(remaining_window, next_read_window_offset);
```
looks overcomplicated, or am I missing something?

Reply:
I just updated the code to simplify the flow. It should not be too complicated now.